deltacat/daft/model.py

```diff
@@ -0,0 +1,258 @@
+from typing import Optional
+
+import pyarrow as pa
+from pyarrow import Field as PaField
+from daft import Schema as DaftSchema, DataType
+from daft.daft import (
+    PartitionField as DaftPartitionField,
+    PartitionTransform as DaftTransform,
+)
+from daft.logical.schema import Field as DaftField
+from daft.io.scan import make_partition_field
+
+from deltacat.storage.model.schema import Schema
+from deltacat.storage.model.interop import ModelMapper
+from deltacat.storage.model.partition import PartitionKey
+from deltacat.storage.model.transform import (
+    BucketingStrategy,
+    Transform,
+    BucketTransform,
+    HourTransform,
+    DayTransform,
+    MonthTransform,
+    YearTransform,
+    IdentityTransform,
+    TruncateTransform,
+)
+
+
+class DaftFieldMapper(ModelMapper[DaftField, PaField]):
+    @staticmethod
+    def map(
+        obj: Optional[DaftField],
+        **kwargs,
+    ) -> Optional[PaField]:
+        """Convert Daft Field to PyArrow Field.
+
+        Args:
+            obj: The Daft Field to convert
+            **kwargs: Additional arguments
+
+        Returns:
+            Converted PyArrow Field object
+        """
+        if obj is None:
+            return None
+
+        return pa.field(
+            name=obj.name,
+            type=obj.dtype.to_arrow_dtype(),
+        )
+
+    @staticmethod
+    def unmap(
+        obj: Optional[PaField],
+        **kwargs,
+    ) -> Optional[DaftField]:
+        """Convert PyArrow Field to Daft Field.
+
+        Args:
+            obj: The PyArrow Field to convert
+            **kwargs: Additional arguments
+
+        Returns:
+            Converted Daft Field object
+        """
+        if obj is None:
+            return None
+
+        return DaftField.create(
+            name=obj.name,
+            dtype=DataType.from_arrow_type(obj.type),  # type: ignore
+        )
+
+
+class DaftTransformMapper(ModelMapper[DaftTransform, Transform]):
+    @staticmethod
+    def map(
+        obj: Optional[DaftTransform],
+        **kwargs,
+    ) -> Optional[Transform]:
+        """Convert DaftTransform to DeltaCAT Transform.
+
+        Args:
+            obj: The DaftTransform to convert
+            **kwargs: Additional arguments
+
+        Returns:
+            Converted Transform object
+        """
+
+        # daft.PartitionTransform doesn't have a Python interface for accessing its attributes,
+        # thus conversion is not possible.
+        # TODO: request Daft to expose Python friendly interface for daft.PartitionTransform
```

Comment:
@rchowell can you help validate my comments here? Please let me know if there is any workaround that I can use to unblock this. If not, what's the process for raising feature request to Daft? Should I directly create an Issue in Daft repo?

Reply:
Yes, related to all three feature requests could please open Daft issue with any additional context? Feel free to group these on one issue since these are all related. I can then work on a proposal to get you some nice APIs. Thanks!

Reply:
Created an Issue on Daft side: https://github.com/Eventual-Inc/Daft/issues/4228